internal/provider/resources/work_pool.go

```diff
@@ -434,3 +401,31 @@ func (r *WorkPoolResource) ImportState(ctx context.Context, req resource.ImportS
 		resource.ImportStatePassthroughID(ctx, path.Root("name"), req, resp)
 	}
 }
+
+// BaseJobTemplatesMatch checks to see if two base_job_template objects are
+// equivalent, accounting for differences in the order of the contents.
+//
+//nolint:ireturn // required by Terraform API
+func BaseJobTemplatesMatch(bjtPlan, bjtReceived map[string]interface{}) diag.Diagnostic {
+	bjtPlanBytes, err := json.Marshal(bjtPlan)
+	if err != nil {
+		return helpers.SerializeDataErrorDiagnostic("base_job_template", "Work Pool base job template", err)
+	}
+
+	bjtReceivedBytes, err := json.Marshal(bjtReceived)
+	if err != nil {
+		return helpers.SerializeDataErrorDiagnostic("base_job_template", "Work Pool base job template", err)
+	}
+
+	opts := jsondiff.DefaultJSONOptions()
+	differenceType, differences := jsondiff.Compare(bjtPlanBytes, bjtReceivedBytes, &opts)
+	if differenceType != jsondiff.FullMatch {
+		return diag.NewAttributeErrorDiagnostic(
+			path.Root("base_job_template"),
+			"Unexpected difference between base_job_template",
+			fmt.Sprintf("Expected the provided base_job_template to be equal to the one retrieved from the API, differences: %s", differences),
```

Comment:
There's an example of what this looks like in the commit message for https://github.com/PrefectHQ/terraform-provider-prefect/pull/249/commits/9f67474fbc35cf4034b0d1f0f21bbadf693f2316.